File: laramie-tesoreria-webapi/src/domain/entities/caja.ts
import CajaAsignacion from "./caja-asignacion";

export default class Caja {

    id: number;
	codigo: string;
	nombre: string;
	orden: number;
	idDependencia: number;
	idEstadoCaja: number;
	idUsuarioActual: number;
	idCajaAsignacionActual: number;
	idRecaudadora: number;

	cajaAsignacion: CajaAsignacion = null;

	constructor(
        id: number = 0,
		codigo: string = "",
		nombre: string = "",
		orden: number = 0,
		idDependencia: number = 0,
		idEstadoCaja: number = 0,
		idUsuarioActual: number = 0,
		idCajaAsignacionActual: number = 0,
		idRecaudadora: number = 0
	)
	{
        this.id = id;
		this.codigo = codigo;
		this.nombre = nombre;
		this.orden = orden;
		this.idDependencia = idDependencia;
		this.idEstadoCaja = idEstadoCaja;
		this.idUsuarioActual = idUsuarioActual;
		this.idCajaAsignacionActual = idCajaAsignacionActual;
		this.idRecaudadora = idRecaudadora;
	}

	setFromObject = (row) =>
	{
        this.id = row.id ?? 0;
		this.codigo = row.codigo ?? "";
		this.nombre = row.nombre ?? "";
		this.orden = row.orden ?? 0;
		this.idDependencia = row.idDependencia ?? 0;
		this.idEstadoCaja = row.idEstadoCaja ?? 0;
		this.idUsuarioActual = row.idUsuarioActual ?? 0;
		this.idCajaAsignacionActual = row.idCajaAsignacionActual ?? 0;
		this.idRecaudadora = row.idRecaudadora ?? 0;
	}

}

File: modules/tesoreria/models/caja.py
from sqlalchemy import Column, BigInteger, String, Integer, ForeignKey
from sqlalchemy.orm import relationship

import sys, os
sys.path.insert(0, os.path.join(os.path.dirname(__file__), "..", ".."))
from shared.database import Base


class Caja(Base):
    __tablename__ = "tesoreria_cajas"

    id = Column(BigInteger, primary_key=True)
    codigo = Column(String(50), nullable=False)
    nombre = Column(String(250), nullable=False)
    orden = Column(Integer, nullable=False, default=0)
    id_dependencia = Column(BigInteger, ForeignKey("tesoreria_dependencias.id"), nullable=True)
    id_estado_caja = Column(BigInteger, nullable=False)
    id_usuario_actual = Column(BigInteger, nullable=True)
    id_caja_asignacion_actual = Column(BigInteger, nullable=True)
    id_recaudadora = Column(BigInteger, nullable=True)

    dependencia = relationship("Dependencia", foreign_keys=[id_dependencia], lazy="joined")
    asignaciones = relationship("CajaAsignacion", back_populates="caja", lazy="dynamic")
